### ISSUE TITLE
Support Python 3.5+ matrix multiplication operator

### ISSUE THREAD

**Issue by mruffalo** [cvxopt/cvxopt#62]
It would be nice if the `cvxopt.matrix` class supported the `@` matrix multiplication operator, which was added to Python 3.5 as per [PEP 465](https://www.python.org/dev/peps/pep-0465/). I've barely started to look through the `cvxopt` source code, but it might be as easy as adding a new method `__matmul__` as an alias to `__mul__`, since the `*` operator is already used for matrix multiplication.

**Comment by martinandersen** [comment]
Thanks for your suggestion. I agree, it should be straightforward, but our main concern is that implementing the new matrix multiplication operator `@` as an alias may lead users to think that `*` performs elementwise multiplication.

**Comment by mruffalo** [comment]
Fair point. I would like to mention that SciPy's sparse matrices also behave in this way, though:

<pre>
Python 3.5.2 (default, Nov 17 2016, 17:05:23) 
Type "copyright", "credits" or "license" for more information.

IPython 5.1.0 -- An enhanced Interactive Python.
?         -> Introduction and overview of IPython's features.
%quickref -> Quick reference.
help      -> Python's own help system.
object?   -> Details about 'object', use 'object??' for extra details.

In [1]: import scipy.sparse

In [2]: A = scipy.sparse.random(5, 5, density=0.5)

In [3]: B = scipy.sparse.random(5, 5, density=0.5)

In [4]: (A @ B).toarray()
Out[4]: 
array([[ 0.15629845,  0.        ,  0.37625768,  0.33491077,  0.10978979],
       [ 0.40734778,  0.0117542 ,  0.37238175,  0.113995  ,  0.3995382 ],
       [ 0.06369844,  0.12954968,  0.15710446,  0.36794765,  0.01906284],
       [ 0.37483914,  0.        ,  0.40246133,  0.46806727,  0.30048457],
       [ 0.32934587,  0.00947863,  0.        ,  0.09920661,  0.32218898]])

In [5]: (A * B).toarray()
Out[5]: 
array([[ 0.15629845,  0.        ,  0.37625768,  0.33491077,  0.10978979],
       [ 0.40734778,  0.0117542 ,  0.37238175,  0.113995  ,  0.3995382 ],
       [ 0.06369844,  0.12954968,  0.15710446,  0.36794765,  0.01906284],
       [ 0.37483914,  0.        ,  0.40246133,  0.46806727,  0.30048457],
       [ 0.32934587,  0.00947863,  0.        ,  0.09920661,  0.32218898]])

In [6]: (_4 == _5).all()
Out[6]: True

In [7]: A.toarray() * B.toarray()
Out[7]: 
array([[ 0.        ,  0.        ,  0.29221804,  0.        ,  0.        ],
       [ 0.2465434 ,  0.        ,  0.14448532,  0.        ,  0.21267772],
       [ 0.        ,  0.        ,  0.        ,  0.08834031,  0.        ],
       [ 0.        ,  0.46565437,  0.        ,  0.        ,  0.        ],
       [ 0.        ,  0.        ,  0.        ,  0.        ,  0.32218898]])

In [8]: A.toarray() @ B.toarray()
Out[8]: 
array([[ 0.15629845,  0.        ,  0.37625768,  0.33491077,  0.10978979],
       [ 0.40734778,  0.0117542 ,  0.37238175,  0.113995  ,  0.3995382 ],
       [ 0.06369844,  0.12954968,  0.15710446,  0.36794765,  0.01906284],
       [ 0.37483914,  0.        ,  0.40246133,  0.46806727,  0.30048457],
       [ 0.32934587,  0.00947863,  0.        ,  0.09920661,  0.32218898]])
</pre>

... so there is precedent for this in the numerical Python ecosystem. I doubt the SciPy developers will ever change this behavior, either, since it would break an enormous amount of working code in the worst possible way (silently wrong results instead of throwing an exception or something).

**Comment by martinandersen** [comment]
Thank you for pointing this out, I wasn't aware of this.

**Comment by kubatucka** [comment]
Any progress on this ? Btw, I love the mention in the PEP : 
"""
The following projects have been alerted to the existence of the PEP, but it's not known what they plan to do if it's accepted (i.e., this is a list of projects whose API fragmentation may or may not be eliminated if this PEP is accepted):

cvxopt

"""
